### DISCUSSION TITLE
How To Accept “Leave This Site” popup

### DISCUSSION THREAD

**Discussion by nsingh-qb** [webdriverio/webdriverio#7191] [Q&A] [answered]
I am new in wdio, my requirement is to handle the leave this site popup. It comes when we navigate to other page without saving unsaved data .
Can someone please help me here how can handle this in wdio.
![image](https://user-images.githubusercontent.com/82230756/127195663-ccbd67ab-78f7-4f4d-891c-5fe976a3f72a.png)

**Comment by christian-bromann** [comment]
Would either of these work:
 - https://webdriver.io/docs/api/webdriver#acceptalert
 - https://webdriver.io/docs/api/webdriver#dismissalert

?

**Reply by nsingh-qb** [reply]
@christian-bromann Thanks for you response , but these I already  tried and it is not working for this popup.
 also by refresh the page that popup is disappearing . but in my case I have to handle this popup like other alert popup (not by refreshing).

**Reply by NavyMint** [reply]
@christian-bromann also have same problem with alerts and accept/dismiss functions not working for me either

**Comment by erwinheitzman** [comment, marked as answer]
You are able to accept / dismiss alerts but what you are seeing isn't an alert but an actual browser popup.

You can disable these using the following:
```js
capabilities: [{
  browserName: 'chrome',
  'goog:chromeOptions': {
    prefs: {
      'profile.managed_default_content_settings.popups' : 1,
      'profile.managed_default_content_settings.notifications' : 1,
    }
  }
}]
```

You can also pass `2` instead of `1` for a different effect. You can research a little about these flags. Hope this helps 👍

**Reply by NavyMint** [reply]
thank you, we will try :)

**Comment by slava-nikulin** [comment]
nothing worked for me :(

**Reply by christian-bromann** [reply]
@slava-nikulin have you tried overwriting the `window.beforeunload` eventhandler? For example:

```ts
await browser.execute(() => {
  window.beforeunload = () => {}
})
```
potentially you need to do this before the app loads, so:

```ts
browser.scriptAddPreloadScript({
  functionDeclaration: `window.beforeunload = () => {}`
})
```

**Reply by slava-nikulin** [reply]
Thanks for the idea!
For some reason the website I test restores beforeunload eventlistener after some nanoseconds. Maybe because it's in react
So I've found a really bad solution imo. But it does its work
```
setInterval(() => {
        window.addEventListener('beforeunload', e => {
          e.stopImmediatePropagation();
        });
      })
```

**Comment by stephen-a2z** [comment]
![image](https://github.com/user-attachments/assets/7db39f14-80f0-4d79-8f69-7203d579ed2b)
I encountered the same issues using selenium
